### PR TITLE
Coinbase_Integration_Fix

### PR DESCRIPTION
**Pull request by hadia**
Fix issues 
- [NMA-1327](https://dashpay.atlassian.net/browse/NMA-1327) Buy Dash Issues
- [NMA-1325](https://dashpay.atlassian.net/browse/NMA-1325)Transfer Issues
- [NMA-1328](https://dashpay.atlassian.net/browse/NMA-1328) Transfer max failed

## How Has This Been Tested?

- [ ] QA (Mobile Team)

## Checklist:

- [ ] I have performed a self-review of my own code and added comments where necessary
- [ ] I have added or updated relevant unit/integration/functional/e2e tests

### CODE REVIEW

File: wallet/build.gradle [hunk 1]
```diff
@@ -213,9 +213,18 @@ android {
             keyAlias "androiddebugkey"
             keyPassword "android"
         }
+        release {
+            keyAlias 'androiddebugkey'
+            keyPassword 'android'
+            storePassword 'android'
+            storeFile file('/Users/hadia/Desktop/debug')
```
[the remark this answers is not in the document]
Comment (hadia) [reply wallet/build.gradle]: Removed

**Review by Syn-McJ — APPROVED** [review]
Looks good

File: integrations/uphold/proguard-rules.pro [hunk 2]
```diff
@@ -2,12 +2,15 @@
...
-
 -keep class org.dash.wallet.integration.liquid.** {
     public protected private *;
 }
+-keep class org.dash.wallet.integration.coinbase_integration.** {
+    public protected private *;
+}
+-keep interface org.dash.wallet.integration.uphold.** {*;}
 -keep interface org.dash.wallet.integration.liquid.** {*;}
+-keep interface org.dash.wallet.integration.coinbase_integration.** {*;}
```
Comment (HashEngineering) [inline integrations/uphold/proguard-rules.pro:13]: Do we need coinbase and liquid items in this file for the uphold module?

File: integrations/coinbase-integration/proguard-rules.pro [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+# hide the original source file name.
+#-renamesourcefileattribute SourceFile
+
+-keepattributes Exceptions, InnerClasses
+-keep class org.dash.wallet.integration.uphold.** {
+    public protected private *;
+}
+-keep class org.dash.wallet.integration.liquid.** {
+    public protected private *;
+}
```
Comment (HashEngineering) [inline integrations/coinbase-integration/proguard-rules.pro:29]: do we need uphold and liquid items in this coinbase file?

File: integrations/liquid/proguard-rules.pro [hunk 4]
```diff
@@ -27,8 +27,12 @@
 -keep class org.dash.wallet.integration.liquid.** {
     public protected private *;
 }
+-keep class org.dash.wallet.integration.coinbase_integration.** {
+    public protected private *;
+}
 -keep interface org.dash.wallet.integration.uphold.** {*;}
```
Comment (HashEngineering) [inline integrations/liquid/proguard-rules.pro:33]: Do we need coinbase and uphold items in the liquid module?  Looks like the previous file had uphold though this is for liquid.

File: wallet/build.gradle [hunk 5]
```diff
@@ -216,6 +216,9 @@ android {
     }
 
     buildTypes {
+        release {
+            signingConfig signingConfigs.release
+        }
```
Comment (HashEngineering) [inline wallet/build.gradle:221]: This needs to be removed as it breaks GitHub Actions.

**Review by HashEngineering — APPROVED** [review]
There are some questions and one required change to fix GitHub Actions.

**Review by HashEngineering — CHANGES_REQUESTED** [review]
https://github.com/dashevo/dash-wallet/pull/995#discussion_r921262493 - fix github actions.

File: wallet/build.gradle (hunk 5 above)
Comment (hadia) [reply wallet/build.gradle]: fixed

File: integrations/liquid/proguard-rules.pro (hunk 4 above)
Comment (hadia) [reply integrations/liquid/proguard-rules.pro]: If i removed  coinbase transfer api  failed

**Review by HashEngineering — APPROVED** [review]
Excellent

**Pull request merged**